csharp/Ice/async/Client.cs

```diff
@@ -15,116 +14,107 @@ public static int Main(string[] args)
+            using (Ice.Communicator communicator = Ice.Util.initialize(ref args, "config.client"))
             {
-                //
-                // Destroy the communicator on Ctrl+C or Ctrl+Break
-                //
-                Console.CancelKeyPress += (sender, eventArgs) => communicator.destroy();
-
-                if(args.Length > 0)
+                if (args.Length > 0)
                 {
-                    Console.Error.WriteLine("too many arguments");
+                    Console.WriteLine("too many arguments");
                     status = 1;
                 }
                 else
                 {
-                    status = run(communicator);
+                    status = Run(communicator);
                 }
             }
         }
-        catch(Exception ex)
+        catch (Exception exception)
         {
-            Console.Error.WriteLine(ex);
+            Console.WriteLine(exception);
             status = 1;
         }
 
         return status;
     }
 
-    private static int run(Ice.Communicator communicator)
+    private static int Run(Ice.Communicator communicator)
     {
-        var hello = HelloPrxHelper.checkedCast(communicator.propertyToProxy("Hello.Proxy"));
-        if(hello == null)
+        HelloPrx hello = HelloPrxHelper.checkedCast(communicator.propertyToProxy("Hello.Proxy"));
+        if (hello == null)
         {
-            Console.Error.WriteLine("invalid proxy");
+            Console.WriteLine("invalid proxy");
             return 1;
         }
 
-        menu();
+        Menu();
 
         string line = null;
         do
         {
             try
             {
-                Console.Out.Write("==> ");
+                Console.Write("==> ");
                 Console.Out.Flush();
                 line = Console.In.ReadLine();
-                if(line == null)
-                {
-                    break;
-                }
-                if(line.Equals("i"))
-                {
-                    hello.sayHello(0);
-                }
-                else if(line.Equals("d"))
-                {
-                    helloAsync(hello);
-                }
-                else if(line.Equals("s"))
-                {
-                    hello.shutdown();
-                }
-                else if(line.Equals("x"))
-                {
-                    // Nothing to do
-                }
-                else if(line.Equals("?"))
+                switch (line)
                 {
-                    menu();
-                }
-                else
-                {
-                    Console.Out.WriteLine("unknown command `" + line + "'");
-                    menu();
+                    case null:
+                        break;
+
+                    case "i":
+                        hello.sayHello(0);
+                        break;
+
+                    case "d":
+                        _ = HelloAsync(hello);
```

Comment:
This doesn't work with Visual Studio 2015, .NET 4.5

We can do `Task.Run(() => HelloAsync(hello));`

Reply:
Fixed, but didn't use Task.Run.